source/modulo_components/modulo_components/component_interface.py

```diff
@@ -318,8 +318,9 @@ def __subscription_callback(self, message: MsgT, attribute_name: str, reader: Ca
         """
         try:
             self.__setattr__(attribute_name, reader(message))
-        except Exception as e:
-            self.get_logger().error(f"Failed to read message for attribute {attribute_name}", throttle_duration_sec=1.0)
+        except (AttributeError, MessageTranslationError) as e:
+            self.get_logger().warn(f"Failed to read message for attribute {attribute_name}: {e}",
```

Comment:
I don't think this should be a `warn` but an `error` as the others.

Reply:
I think warn is fine here. We will have to write the general guidelines for this and other things, but I would suggest we leave `error` for cases where something fails and leave the component in a different state than expected. For example,
- a user-requested transition callback or service fails
- an run-time exception cannot be handled and `raise_error` is called

I would suggest warn is more appropriate for periodic things:
- subscription callback cannot read the type
- on step callback is interrupted (but the exception is handled and suppressed)

The difference with `warn` is that the component will still be running and in the same state as before, but we want to log the warning that something is not right.